### PR TITLE
New games!

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Motivation for this change

blockattack: init at 2.7.0
the-legend-of-edgar: init at 1.35

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by L-as** [comment]
Shouldn't this be 2 PRs?

**Comment by AndersonTorres** [comment]
> Shouldn't this be 2 PRs?

Multiple PRs is the most recommended approach, to be sure. However, I prefer to upload small, straightforward programs in batches. This is not a hard thing to follow here

File: pkgs/games/blockattack/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+, pkg-config
+, zip
+}:
+
+stdenv.mkDerivation rec {
+  pname = "blockattack";
+  version = "2.7.0";
+
+  src = fetchFromGitHub {
+    name = "${pname}-${version}";
```
Comment (legendofmiracles) [inline pkgs/games/blockattack/default.nix:21]: What does the name attribute do here?

File: pkgs/games/the-legend-of-edgar/default.nix [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  makeFlags = [
+    "PREFIX=${placeholder "out"}"
+    "BIN_DIR=${placeholder "out"}/bin/"
+  ];
+
+  # TODO: remove the setting below when the next version arrives
+  # https://github.com/riksweeney/edgar/pull/57
+  preBuild = ''
+    export CFLAGS=`sdl2-config --cflags`
```
Comment (legendofmiracles) [inline pkgs/games/the-legend-of-edgar/default.nix:52]:
```suggestion
    export CFLAGS=$(sdl2-config --cflags)
```
Generally preferred

**Review by legendofmiracles — COMMENTED** [review]
The mark as broken commits should be squashed

File: pkgs/games/blockattack/default.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/games/blockattack/default.nix:21]: Completely optional, but helpful. It is better than read `/nix/store/giantHash-source` in the logs.

**Pull request merged**